pkg/name/registry.go

```diff
@@ -50,6 +50,11 @@ func (r Registry) String() string {
 	return r.Name()
 }
 
+// Repo returns a Repository in the Registry with the given name.
+func (r Registry) Repo(repo string) Repository {
```

Comment:
This idea might be bad, but hear me out: consider making this variadic.

I'm in a weird spot with `registry.k8s.io` where I was to call `google.List` on `registry.k8s.io`, but I can't because it doesn't parse. They redirect to AR but with an included path, but effectively calling list on `registry.k8s.io` is ~equivalent to `/v2/_catalog` because of GCR's custom response.

If we make this variadic, then `Reference.Repo()` could implement the same interface. So could `Repository.Repo(...)`. Essentially we get a way to either get the repo for a given thing _or_ get a child repo for a given thing... wdyt?

The variadic is kinda weird for ref, but if (for some reason) you wanted to get a child repo from a ref, instead of doing `ref.Context().Repo("child") or `ref.Repo().Repo("child")`, you could just do `ref.Repo("child")`.

For `len(repos) > 1`, I think `path.Join`ing them is reasonable.

Reply:
Is there a difference between `Repo("a", "b", "c")` and `Repo(path.Join("a", "b", "c"))`? It's more verbose but clearer IMO.

I'm not sure it's clear what `NewTag("reg.io/repo:tag").Repo("child")` does -- does it give you `reg.io/repo/child:tag`, or strip the tag? Same for digest. `NewTag("reg.io/repo:tag").Context().Repo("child")` is more verbose but clearer, because I'm explicitly stepping up into the repo, and leaving the tag behind.

I'm fine with `Repository` having `Repo(child string)`. I don't have an immediate need for it personally, but I could see it. Making that variadic seems like it's maybe trying to be too clever. If we want to make it variadic we can do that later, so long as it's not part of an interface like `Reference`.